### ISSUE TITLE
Error when calling .Convert() on Property when using empty Map()

### ISSUE THREAD

**Issue by mikegwyn17** [JoshClose/CsvHelper#2118]
**Describe the bug**
```
dotnet --info
.NET SDK:
Version:   7.0.101
Commit:    bb24aafa11
```
CSVHelper Version: 30.0.1

When using a ClassMap for writing a CSV. If the field being mapped is not selected from the model, when accessing it in the .Convert() method an exception is thrown. This only happens when the Map() is called without a lambda, when called with a field selected it doesn't error, but this doesn't allow multiple values to be mapped from the same field.

```csharp
Unhandled exception. CsvHelper.WriterException: An unexpected error occurred. See inner exception for details.
IWriter state:
   Row: 2
   Index: 0
   HeaderRecord:
2
```

**To Reproduce**
```csharp
using System.Globalization;
using CsvHelper;
using CsvHelper.Configuration;


var records = new List<Foo>
{
    new() { Bar = new HashSet<string> { "foo" } },
    new() { Bar = new HashSet<string> { "bar" } },
};

await using var writer = new StreamWriter(Path.Combine(Directory.GetCurrentDirectory(), "test.csv"));
var config = new CsvConfiguration(CultureInfo.InvariantCulture);
await using var csv = new CsvWriter(writer, config);
csv.Context.RegisterClassMap<FooMap>();
await csv.WriteRecordsAsync(records);

public class Foo
{
    public HashSet<string> Bar { get; set; }
}

public sealed class FooMap : ClassMap<Foo>
{
    public FooMap()
    {
        Map().Index(0).Name("Foo").Convert(x => (x.Value as Foo).Bar.Contains("foo") ? "True" : "False");
        Map().Index(1).Name("Bar").Convert(x => (x.Value as Foo).Bar.Contains("bar") ? "True" : "False");
    }
}
```

This code works in a previous version of CSVHelper (19.0.1)
```csharp
using System.Globalization;
using CsvHelper;
using CsvHelper.Configuration;


var records = new List<Foo>
{
    new() { Bar = new HashSet<string> { "foo" } },
    new() { Bar = new HashSet<string> { "bar" } },
};

await using var writer = new StreamWriter(Path.Combine(Directory.GetCurrentDirectory(), "test.csv"));
var config = new CsvConfiguration(CultureInfo.InvariantCulture);
config.RegisterClassMap<FooMap>();
await using var csv = new CsvWriter(writer, config);
await csv.WriteRecordsAsync(records);

public class Foo
{
    public HashSet<string> Bar { get; set; }
}

public sealed class FooMap : ClassMap<Foo>
{
    public FooMap()
    {
        Map().Index(0).Name("Foo").ConvertUsing(x => (x as Foo).Bar.Contains("foo") ? "True" : "False");
        Map().Index(1).Name("Bar").ConvertUsing(x => (x as Foo).Bar.Contains("bar") ? "True" : "False");
    }
}
```
**Expected behavior**
I expect the ClassMap to map the same field to multiple values without throwing an exception, without needing to select the field in the Map().

**Additional context**
We are upgrading from 19.0.1 to 30.0.1

```chsarp
/Users/mikegwyn/RiderProjects/CSVHelperBug/CSVHelper30/bin/Debug/net7.0/CSVHelper30
Unhandled exception. CsvHelper.WriterException: An unexpected error occurred. See inner exception for details.
IWriter state:
   Row: 2
   Index: 0
   HeaderRecord:
2

 ---> System.ArgumentException: Expression of type 'CsvHelper.ConvertToStringArgs`1[Foo]' cannot be used for parameter of type 'CsvHelper.ConvertToStringArgs`1[System.Object]' (Parameter 'arg0')
   at System.Dynamic.Utils.ExpressionUtils.ValidateOneArgument(MethodBase method, ExpressionType nodeKind, Expression arguments, ParameterInfo pi, String methodParamName, String argumentParamName, Int32 index)
   at System.Linq.Expressions.Expression.Invoke(Expression expression, Expression arg0)
   at System.Linq.Expressions.Expression.Invoke(Expression expression, IEnumerable`1 arguments)
   at CsvHelper.Expressions.ObjectRecordWriter.CreateWriteDelegate[T](T record)
   at CsvHelper.Expressions.RecordWriter.GetWriteDelegate[T](T record)
   at CsvHelper.Expressions.RecordWriter.Write[T](T record)
   at CsvHelper.Expressions.RecordManager.Write[T](T record)
   at CsvHelper.CsvWriter.WriteRecord[T](T record)
   --- End of inner exception stack trace ---
   at CsvHelper.CsvWriter.WriteRecord[T](T record)
   at CsvHelper.CsvWriter.WriteRecordsAsync[T](IEnumerable`1 records, CancellationToken cancellationToken)
   at Program.<Main>$(String[] args) in /Users/mikegwyn/RiderProjects/CSVHelperBug/CSVHelper30/Program.cs:line 16
   at Program.<Main>$(String[] args) in /Users/mikegwyn/RiderProjects/CSVHelperBug/CSVHelper30/Program.cs:line 16
   at Program.<Main>$(String[] args) in /Users/mikegwyn/RiderProjects/CSVHelperBug/CSVHelper30/Program.cs:line 16
   at Program.<Main>(String[] args)
```

**Comment by Rob-Hague** [comment]
A fix is at #2119 

Workarounds:

1.
```diff
- await csv.WriteRecordsAsync(records);
+ await csv.WriteRecordsAsync((IEnumerable)records);
```

2.
```diff
- var records = new List<Foo>
+ var records = new List<object>
```

**Comment by mikegwyn17** [comment]
@Rob-Hague thanks for the quick response and the workaround.
